modules/schema/src/main/java/org/apache/ignite/internal/util/ColocationHashFunction.java

```diff
@@ -15,20 +15,17 @@
  * limitations under the License.
  */
 
-package org.apache.ignite.internal.sql.engine.metadata;
-
-import java.util.function.ToIntFunction;
+package org.apache.ignite.internal.util;
 
 /**
- * AffinityService interface.
- * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
+ * Function to calculate a hash of the colocation fields of a row.
  */
-public interface AffinityService {
+@FunctionalInterface
+public interface ColocationHashFunction<T> {
```

Comment:
Why do we need such interface in a Schema module? To be honest, I don't get the idea

Reply:
Why do we need to _mark_ all implementation? Why do we need several implementations anyway? Actually we need exactly one implementation of such function. Furthermore, the function should extract partition id, not the abstract hash.

Reply:
_> Why do we need several implementations anyway?_
Because currently we have different implementations for BinaryRow and RowT.

My proposal  -  remove `ColocationHashFunction` interface and keep hash calculations for `BinaryRow` and `RowT` independent (as anonymous classes). They will both use HashCalculator, maybe I was wrong and we don't need to unify them any more.

WDYT?